### PR TITLE
Ensure cookies are returned for 3xx redirects as well

### LINKED ISSUE #3758: Shadow Endpoints: slight problems with set-cookie and other headers
### Describe the bug

Firstly, thank you for shadow endpoints! They are a lovely simplification, and make perfect sense when implementing straightforward pages.

I've noticed two things which aren't quite right with the processing of the shadow endpoint's Response:

### Reproduction

**1. `set-cookie` is assumed to be an array**
See https://github.com/sveltejs/kit/commit/647131e00149cb5299042eedeb880efcd855a2df#diff-7266c49c71803f38dda70345d857f13a7a2494cc649b70af6b689627199f5b79R404-R406

<details>
 <summary>Reproduction: string is exploded</summary>

**endpoint.js**

```js
export async function post() {
    return {
        headers: {
            'set-cookie': 'a=b',
        },
    };
}
```
**test**
```sh
bash-5.1$ curl -X POST -I http://localhost:5001/endpoint

HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
content-type: text/html
etag: "l9x8dd"
permissions-policy: interest-cohort=()
set-cookie: a
set-cookie: =
set-cookie: b
Date: Mon, 07 Feb 2022 10:16:38 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```
</details>

<details>
 <summary>Reproduction: array is handled correctly</summary>

**endpoint.js**
```js
export async function post() {
    return {
        headers: {
            'set-cookie': ['a=b'],
        },
    };
}
```

**test**
```sh
bash-5.1$ curl -X POST -I http://localhost:5001/endpoint

HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
content-type: text/html
etag: "l9x8dd"
permissions-policy: interest-cohort=()
set-cookie: a=b
Date: Mon, 07 Feb 2022 10:16:38 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```
</details>

<details>
 <summary>Expected: string is treated as a singular header</summary>

**endpoint.js**
```js
export async function post() {
    return {
        headers: {
            'set-cookie': 'a=b',
        },
    };
}
```

**test**
```sh
bash-5.1$ curl -X POST -I http://localhost:5001/endpoint

HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
content-type: text/html
etag: "l9x8dd"
permissions-policy: interest-cohort=()
set-cookie: a=b
Date: Mon, 07 Feb 2022 10:16:38 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```
</details>

**2. `set-cookie` and other headers aren't preserved for 3xx redirects**
See https://github.com/sveltejs/kit/commit/647131e00149cb5299042eedeb880efcd855a2df#diff-7266c49c71803f38dda70345d857f13a7a2494cc649b70af6b689627199f5b79R408-R416

Note that only `headers.location` is copied across in the case of a 3xx redirect.

<details>
 <summary>Reproduction: `set-cookie` is ignored for 307 redirect</summary>

**endpoint.js**
```js
export async function post() {
    return {
        status: 307,
        headers: {
            location: '/',
            'set-cookie': ['a=b'],
        },
    };
}

```

**test**
```sh
bash-5.1$ curl -X POST -I http://localhost:5001/endpoint

HTTP/1.1 307 Temporary Redirect
Access-Control-Allow-Origin: *
location: /
Date: Mon, 07 Feb 2022 10:25:09 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```
</details>

<details>
 <summary>Expected: `set-cookie` is preserved for 307 redirect</summary>

**endpoint.js**
```js
export async function post() {
    return {
        status: 307,
        headers: {
            location: '/',
            'set-cookie': ['a=b'],
        },
    };
}

```

**test**
```sh
bash-5.1$ curl -X POST -I http://localhost:5001/endpoint

HTTP/1.1 307 Temporary Redirect
Access-Control-Allow-Origin: *
location: /
set-cookie: a=b
Date: Mon, 07 Feb 2022 10:25:09 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```
</details>

### Logs

_No response_

### System Info

```shell
System:
    OS: Linux 5.10 Alpine Linux
    CPU: (6) arm64 unknown
    Memory: 10.94 GB / 15.60 GB
    Container: Yes
    Shell: 1.34.1 - /bin/ash
  Binaries:
    Node: 16.13.1 - /usr/local/bin/node
    Yarn: 1.22.15 - /usr/local/bin/yarn
    npm: 8.1.2 - /usr/local/bin/npm
  npmPackages:
    @sveltejs/adapter-node: ^1.0.0-next.67 => 1.0.0-next.67
    @sveltejs/kit: ^1.0.0-next.260 => 1.0.0-next.260
    svelte: ^3.46.2 => 3.46.2
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by georgecrawford**
Fixes #3758 (in part)

The block which handles 3xx redirects wasn't returning the `cookies` property. This PR ensures that 3xx responses from a shadow endpoint return cookies as expected.

If the change looks OK, I'll add tests to cover it.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by georgecrawford** [comment]
I've done my best with adding tests, but definitely need some direction from the project maintainers about whether the approach is correct, and why one of the tests fails. Thanks for any pointers!

File: packages/kit/test/apps/basics/test/test.js [hunk 1]
```diff
@@ -346,7 +346,7 @@ test.describe.parallel('CSS', () => {
 	});
 });
 
-test.describe.parallel('Shadowed pages', () => {
+test.describe.parallel.only('Shadowed pages', () => {
```
Comment (Rich-Harris) [inline packages/kit/test/apps/basics/test/test.js:349]:
```suggestion
test.describe.parallel('Shadowed pages', () => {
```

Comment (georgecrawford) [reply packages/kit/test/apps/basics/test/test.js]: Hah, thanks Rich

**Comment by Rich-Harris** [comment]
I found the reason for the failing test — when we request shadow endpoint data from the client, redirects are opaque, so we respond with a 204 with a `x-sveltekit-location` header instead. Other headers were getting discarded

**Comment by Rich-Harris** [comment]
thank you!

**Pull request merged**